### DISCUSSION TITLE
few questions with GroupCog

### DISCUSSION THREAD

**Discussion by keqking** [Rapptz/discord.py#8249] [Q&A]
Hi,

I have few questions about ext.commands.GroupCog and how to use it after reading [its docs](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html?highlight=groupcog#discord.ext.commands.Cog).

1. Since docs for it mentions, GroupCog represents/inherits from ext.commands.Cog that also doubles as a parent app_commands.Group, does it mean that it basically offers same functionality as hybrid group? For example, my group cog has 2 cmds: `[p]foo` and `[p]bar` and with GroupCog (and defining `group_name='test'` during subclassing), it creates slash cmds `/test foo` and `/test bar` too?
1.1. or GroupCog is only usable for app commands only? no ext text commands support?

2. how does cmd signature look for cmd objects/functions inside GroupCog?
2.1. does cmds inside GroupCog have access to `ctx` or Context arg since it inherits from ext.commands.Cog I would assume it would?
2.2. which of these would be valid cmd signature in this context?
```py
from discord.ext import commands


class Test(commands.GroupCog, group_name='test'):

    @commands.command()
    async def foo(self, ctx, input: str):
        ...
```
or
```py
from discord import app_commands
from discord.ext import commands


class Test(commands.GroupCog, group_name='test'):

    @app_commands.command()
    async def foo(self, interaction, input: str):
        ...
```
I saw some tags in dpy server which used the 2nd example and it felt confusing that cmds inside GroupCog doesnt have access to `ctx` since GroupCog inherits from ext.commands.Cog? 🤔

It would be great if example in docs for GroupCog can be expanded bit more if possible. 😅

**Comment by Rapptz** [comment]
1. `GroupCog` only affects the application commands. So if you have `/foo` and `/bar` they become `/test foo` and `/test bar` but the text commands remain `[p]foo` and `[p]bar`.
2. `Command.signature` is not impacted. If you mean the function signature then that's not impacted either.
   1. Commands function as normal, so they accept context.
   2. Both are fine. Text and hybrid commands accept context, app commands accept interaction.

**Reply by keqking** [reply]
oh that answers my question! thank you alot! ❤️
